### PR TITLE
Here I go again on my Chown

### PR DESCRIPTION
**Pull request by cfunkhouser**
Add a Chown function matching the os.Chown signature to afero.Fs, and implement it in the various Fs implementations.

### CODE REVIEW

**Comment by cfunkhouser** [comment]
Fixes #122

**Comment by corentone** [comment]
@cfunkhouser your timing is perfect, I'm also looking for this!
I'm not familiar with the internal of afero but I'd be happy to help.

File: mem/file.go [hunk 1]
```diff
@@ -81,6 +83,14 @@ func SetMode(f *FileData, mode os.FileMode) {
 	f.mode = mode
 }
 
+func SetUID(f *FileData, uid int) {
+	f.uid = uid
```
Comment (mbertschler) [inline mem/file.go]: This and the SetGID function will need locks, like they were recently added to master to not have races on the FileData struct.

File: memmap.go [hunk 2]
```diff
@@ -347,6 +347,24 @@ func (m *MemMapFs) Chtimes(name string, atime time.Time, mtime time.Time) error
...
+	name = normalizePath(name)
+
+	m.mu.RLock()
+	f, ok := m.getData()[name]
+	m.mu.RUnlock()
+	if !ok {
+		return &os.PathError{Op: "chown", Path: name, Err: ErrFileNotFound}
+	}
+
+	m.mu.Lock()
```
Comment (mbertschler) [inline memmap.go]: If the SetUID and SetGID functions are using locks, we won't need this lock anymore.

**Comment by kargakis** [comment]
Needs a rebase

**Comment by kargakis** [comment]
Feel free to reopen if you plan to fix.

**Pull request closed without merging**